technical_architecture/performances_guide/batch_page_size.rst

```diff
@@ -0,0 +1,13 @@
+Job product batch size
+======================
+
+A product catalog is unique to one's need. It is configured with a custom number of attributes, families, locales, channels, etc. Those different combinations may affect the batch jobs (CSV imports/exports, mass edits, indexation, etc) performances.
+
+Indeed the size of the products will define how much RAM they will take to be processed. During a product batch jobs, we regularly flush the processed products into the database to avoid too much memory consumption.
+
+But, if we flush them too often, the transaction time can slow the process: for each iteration, we clear the doctrine cache (to free some memory), index the products (so more communications between PH and elastic search) and update the mysql database. On the other hand, if we the flush are too separated in time, the RAM consumption can increase to a point where the PHP's garbage collector is called too many times and slows down the batch jobs.
```

Comment:
What is PH ? Maybe you mean PIM?

Reply:
"if we the flush are too separated in time" => "if the flushes are too separated in time"

Reply:
```suggestion
But, if we flush them too often, the transaction time can slow the process: for each iteration, we clear the doctrine cache (to free some memory), index the products (so more communications between PHP and elastic search) and update the mysql database. On the other hand, if we the flush are too separated in time, the RAM consumption can increase to a point where the PHP's garbage collector is called too many times and slows down the batch jobs.
```

Reply:
I meant PHP